agent/php_execute.c

```diff
@@ -1715,48 +1823,143 @@ static void nr_php_instrument_func_begin(NR_EXECUTE_PROTO) {
+       * segment.
+       */
+
+      if (nrunlikely(NULL != NRPRG(uncaught_exception))) {
+        /*
+         * First check if it's the root because obviously, prev_execute won't
+         * exist.
+         */
+        if (NRTXN(segment_root) != NRTXN(force_current_segment)) {
+          /*
+           * Get the current segment if it exists.
+           */
+          nr_segment_t* exception_segment = NRTXN(force_current_segment);
+          if (NULL != exception_segment) {
+            /*
+             * If the metadata info doesn't match the previous callers This,
+             * then we know the uncaught exception occurred which caused the
+             * fcall_end function to not be called.  Clean up dangling stacked
+             * segments.
+             */
+            nr_php_execute_metadata_t* md = exception_segment->metadata;
+            if ((NULL != md)
+                && (md->execute_data_this
+                    != &execute_data->prev_execute_data->This)) {
+              /*
+               * Close all previous segments, attaching the uncaught exception
+               * as necessary.
+               */
+              nr_php_observer_exception_segments_end(
+                  NRPRG(uncaught_exception),
+                  &execute_data->prev_execute_data->This);
+            }
+          }
+          php_observer_clear_uncaught_exception_globals();
+        }
+      }
+    }
+
     segment = nr_php_stacked_segment_init(segment);
     if (nrunlikely(NULL == segment)) {
       nrl_verbosedebug(NRL_AGENT, "Error initializing stacked segment.");
       return;
     }
-    segment->txn_start_time = nr_txn_start_time(NRPRG(txn));
-    segment->wraprec = wraprec;
-    segment->lineno = nr_php_zend_execute_data_lineno(execute_data);
-    zcaught = nr_zend_call_oapi_special_before(wraprec, segment,
-                                               NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
-    if (nrunlikely(zcaught)) {
-      zend_bailout();
+
+    nr_php_observer_metadata_init(segment->metadata, NR_EXECUTE_ORIG_ARGS);
+    metadata = segment->metadata;
+    metadata->execute_data_this = &execute_data->This;
+    /*
+     * Metadata deinit is handled when the segment is destroyed.
+     */
+
+    if (NULL != wraprec) {
+      /*
+       * If a function needs to have arguments modified this is the place.
+       */
+      segment->wraprec = wraprec;
+      zcaught = nr_zend_call_oapi_special_before(wraprec, segment,
+                                                 NR_EXECUTE_ORIG_ARGS);
+      if (nrunlikely(zcaught)) {
+        zend_bailout();
+      }
+
+      nr_txn_force_single_count(NRPRG(txn), wraprec->supportability_metric);
+      /*
+       * Check for, and handle, frameworks.
+       */
+      if (wraprec->is_names_wt_simple) {
+        nr_txn_name_from_function(NRPRG(txn), wraprec->funcname,
+                                  wraprec->classname);
+      }
     }
   }
   return;
```

Comment:
Since this function is void and nothing happens after this point, this `return` can be removed
```suggestion
```

Reply:
Please see: 
[fed7f9c](https://github.com/newrelic/newrelic-php-agent/pull/580/commits/fed7f9c6c7a980c3d8bb1d87f43f9218363fc105)

and 

[4c64593](https://github.com/newrelic/newrelic-php-agent/pull/580/commits/4c645933f57d20c7395153e36b409cc2358b0463)